client/components/theme/index.jsx

```diff
@@ -172,7 +176,7 @@ const Theme = React.createClass( {
 								context: 'singular noun, the currently active theme'
 							} ) }</span>
 						}
-						{ price && ! purchased &&
+						{ config.isEnabled( 'jetpack/pijp' ) && ! this.props.isJetpack && price && ! purchased &&
```

Comment:
This will hide the price in all environments not in the feature flag, regardless of Jetpack status